code/z_adventurezones/safehouse.dm

```diff
@@ -0,0 +1,708 @@
+		D.CopyOther(M.bioHolder)
+
+		for_by_tcl(O, /obj/machinery/handscanner/bio_handscanner)
+			O.allowed_bioHolders = D.Uid //Copy the Uid only, copying and comparing against all bioHolder data is too prone to error.
+
+		for_by_tcl(O, /obj/item/reagent_containers/iv_drip/dead_exec)
+			if(!O.reagents.has_reagent("blood"))
+				return
+			var/datum/reagent/blood/B = O.reagents.reagent_list["blood"]
+			B.data = D //Give the blood Jean's bioHolder info.
+
+		SPAWN_DBG(5 SECONDS) //Jean's just here to set up the puzzle, we don't want him sticking around.
+		qdel(M)
+
+	attack_hand(mob/user as mob)
+		boutput(user, "An advanced cloning pod, designed to be operated automatically through packets. What a great idea!<br>Currently idle.<br><span class='alert'>Biomatter reserves are depleted.</span>")
+		playsound(src.loc, "sound/impact_sounds/Generic_Stab_1.ogg", 25, 1)
+		src.add_fingerprint(user)
+		return
+
+// CLONESCAN HEALTH IMPLANT
+
+/obj/item/implant/health/exp_health_implant
+	name = "experimental health implant"
+	desc = ""
+	icon_state = "implantpaper-b"
+	impcolor = "b"
+	var/clonescanned = 0
+
+	activate()
+		..()
+		clonescanned = 0
+
+	do_process(var/mult = 1) //On do_process so we don't have to worry if the user is dead / in ghostVR.
+		if(!ishuman(src.owner))
+			return
+		var/mob/living/carbon/human/H = src.owner
+		if(clonescanned < 11)
+			switch(clonescanned++)
+				if(1)
+					boutput(H,"Life signs detected. Initiating scanning procedure.")
+				if(4)
+					boutput(H,"Scan complete. Contacting remote server for data upload.")
+				if(7)
+					boutput(H,"ERROR: Remote server unavailable. Commencing emergency transmisssion protocols. Broadcasting data on all known frequencies.")
+				if(10)
+					for_by_tcl(C, /obj/machinery/computer/cloning)
+						C.scan_mob(H)
+						var/datum/data/record/R = new /datum/data/record()
+						R = C.find_record(H.ckey)
+						if(!isnull(R))
+							boutput(H,"Link to cloning computer establised succesfully.")
+							playsound(src.loc, 'sound/machines/ping.ogg', 50, 1)
+							if(prob(20) && !H.traitHolder.hasTrait("puritan")) // Too uncommon to weaponise but common enough to deter general usage.
+								R["fields"]["traits"] += "puritan" // Did the player learn nothing from the prefab??
+
+	on_death()
+		if(!ishuman(src.owner))
+			return
+		for_by_tcl(C, /obj/machinery/computer/cloning)
+			var/mob/living/carbon/human/H = src.owner
+			C.scan_mob(H)
+			var/R = C.find_record(H.ckey)
+			if(R)
+				playsound(src.loc, 'sound/machines/ping.ogg', 50, 1)
+				if(prob(20) && !H.traitHolder.hasTrait("puritan")) // Too uncommon to weaponise but common enough to deter general usage.
+					R["fields"]["traits"] += "puritan" // Did the player learn nothing from the prefab??
+
+//DECORATIVE OBJECTS
+
+/obj/decal/fakeobjects/beacon
+	name = "tracking beacon"
+	desc = "A small tracking beacon. What's it doing all the way out here?"
+	icon = 'icons/obj/items/device.dmi'
+	icon_state = "beacon"
+	anchored = 1
```

Comment:
maybe some kind of busted beacon, so it's clear why you can't set a teleporter to it?

Reply:
Object description & sprite amended to better set player expectations.